router/http.go

```diff
@@ -275,62 +303,28 @@ func (s *HTTPListener) findRouteForHost(host string) *httpRoute {
 	return nil
 }
 
-func fail(sc *httputil.ServerConn, req *http.Request, code int, msg string) {
-	resp := &http.Response{
-		StatusCode:    code,
-		ProtoMajor:    1,
-		ProtoMinor:    0,
-		Request:       req,
-		Body:          ioutil.NopCloser(bytes.NewBufferString(msg)),
-		ContentLength: int64(len(msg)),
-	}
-	sc.Write(req, resp)
+func fail(w http.ResponseWriter, code int, msg string) {
+	w.Header().Set("Content-Length", strconv.Itoa(len(msg)))
```

Comment:
It does force a fixed-length (non-chunked) response. As far as whether we want to `Connection: close`, after an error, that's generally a no, but there may be cases (specifically most `fail()` calls inside the TCP proxying path) where we do want to close the conn.

We can either add a param to `fail()` to indicate that, or just call `w.Header().Set("Connection", "close")` in each specific place where we want to do that. Extra param sounds better to me.